checkstyle/src/main.rs

```diff
@@ -29,7 +31,9 @@ fn main() -> Result<(), std::io::Error> {
 
     let file_paths = matching_files(&target_path, args.include, args.exclude);
 
-    let nonmatching_files = nonmatching_files(&regex_file_path, file_paths);
+    let nonmatching_files = nonmatching_files(&regex_file_path, file_paths.clone());
+
+    let end_empty_line = !no_empty_line(file_paths.clone()).is_empty();
```

Comment:
A good sign we've got this composition right is that we can return many of the errors at once. Instead, let's do the same for `end_empty_line` as we do for `nonmatching_files` and take the list of files that don't comply.

```suggestion
    let nonmatching_files = nonmatching_files(&regex_file_path, file_paths.clone());

    let end_empty_line = !no_empty_line(file_paths.clone());
```